scripts/tools/file_size_from_nm.py

```diff
@@ -660,6 +708,79 @@ def symbols_from_nm(elf_file: str) -> list[Symbol]:
     return symbols
 
 
+def fetch_symbols(elf_file: str, fetch: FetchStyle) -> Tuple[list[Symbol], str]:
+    """Returns the sumbol list and the separator used to split symbols
+    """
+    if fetch == FetchStyle.NM:
+        return symbols_from_nm(elf_file), "::"
+    else:
```

Comment:
super nit, do you want to also do assert here that `fetch == FetchStyle.OBJDUMP`? I only ask because I see you did something similar above

Reply:
Added assert ... however then figured we are on python 3.10+ anyway, so switched to `match` for readability.